nifi-extension-bundles/nifi-extension-utils/nifi-event-transport/src/main/java/org/apache/nifi/event/transport/netty/EventLoopGroupFactory.java

```diff
@@ -35,6 +34,16 @@ class EventLoopGroupFactory {
 
     private int workerThreads;
 
+    private NettyTransports.NettyTransport nettyTransport;
+
+    public EventLoopGroupFactory() {
+        this(NettyTransports.getDefaultNettyTransport());
+    }
+
+    public EventLoopGroupFactory(NettyTransports.NettyTransport nettyTransport) {
+        this.nettyTransport = nettyTransport;
+    }
```

Comment:
I recommend avoiding the addition of the parameterized constructor and instead use the availability determination logic in the get default method. Callers should not need to select a particular implementation, thus keeping the selection details private to this factory class.

Reply:
Agreed.  I'll remove the parameterized constructor.